modules/Bio/EnsEMBL/Utils/IO/BEDSerializer.pm

```diff
@@ -118,11 +140,29 @@ sub _write_Transcript {
 
     #Same again but for the end
     $coding_end = $self->_cdna_to_genome($new_transcript, $cdna_end);
+
+    #Also figure out complete 5' and 3' CDS tags
+    my ($five_prime_nc, $three_prime_nc) =  map { $_->[0] } 
+                                            grep { defined $_ && @{$_} } 
+                                            map { $transcript->get_all_Attributes($_) } 
+                                            qw/cds_start_NF cds_end_NF/; 
+
+    ($cds_start_status, $cds_end_status) = ('cmpl', 'cmpl');
+    if($transcript->strand() == -1) {
+      # reverse the calls when on the negative strand
+      $cds_start_status = 'incmpl' if $three_prime_nc;
+      $cds_end_status = 'incmpl' if $five_prime_nc;
+    }
+    else {  
+      $cds_start_status = 'incmpl' if $five_prime_nc;
+      $cds_end_status = 'incmpl' if $three_prime_nc;
+    }
   }
   else {
     # apparently looking at UCSC's own BED output formats we do not need to bother
     # coverting $coding_start into 0 based coords for this one ... odd
-    $coding_start = $new_transcript->seq_region_end();
+    $coding_start = $new_transcript->seq_region_start();
+    $coding_start--;
     $coding_end = $coding_start;
```

Comment:
That comment might be misleading now looking at some other outputs ucsc has. The format is ambiguous as to what a column should contain beyond it being the data point in question. 

Sent from my mobile.

> On 17 Jul 2015, at 12:05, Magali Ruffier <notifications@github.com> wrote:
> 
> In modules/Bio/EnsEMBL/Utils/IO/BEDSerializer.pm:
> 
> >    }
> >    else {
> >      # apparently looking at UCSC's own BED output formats we do not need to bother
> >      # coverting $coding_start into 0 based coords for this one ... odd
> > -    $coding_start = $new_transcript->seq_region_end();
> > +    $coding_start = $new_transcript->seq_region_start();
> > +    $coding_start--;
> >      $coding_end = $coding_start;
> does that mean we do have to convert to 0-based coordinates?
> 
> —
> Reply to this email directly or view it on GitHub.
>